### PR TITLE
Remove set_default_dtype from jit_metaprogramming_utils

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#53621 Remove set_default_dtype from jit_metaprogramming_utils**
* #53617 Marginally improve pytest collection for top-level test files
* #53611 Fix inaccurate dispatch table for fill_
* #53610 Convert a few more checks to raise NotImplementedError

I did this by inlining the setter to the various imports.  Having
done this, I removed the float set from test_linalg.py

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D26918375](https://our.internmc.facebook.com/intern/diff/D26918375)

### CODE REVIEW

File: test/test_ops.py [hunk 1]
```diff
@@ -18,6 +18,8 @@
     check_alias_annotation
 from torch.testing._internal.jit_utils import disable_autodiff_subgraph_inlining
 
+torch.set_default_dtype(torch.double)
```
Comment (mruberry) [inline test/test_ops.py:21]: Is this actually required?

**Comment by mruberry** [comment]
It looks like there's one internal-only file that also imports this.

**Review by mruberry — APPROVED** [review]
Cool; it'd be nice if we could remove the default dtype setting from test_ops.py, too, though (see inline comment).

There may be an internal-only change needed for this PR, too.

**Comment by ezyang** [comment]
Looks like this PR is wrong lol. I guess I'm gonna have to investigate more.

**Pull request closed without merging**